operators/pkg/forge/services.go

```diff
@@ -51,14 +51,30 @@ func ServiceSpec(instance *clv1alpha2.Instance, environment *clv1alpha2.Environm
 		ports = append(ports, serviceSpecTCPPort(SSHPortName, SSHPortNumber))
 	}
 
-	// Add the desktop port only if enabled.
+	// Add the GUI port only if enabled.
 	if environment.GuiEnabled {
 		ports = append(ports, serviceSpecTCPPort(GUIPortName, GUIPortNumber))
 	}
 
-	// Add the "MyDrive" port only if the environment is a container.
-	if environment.EnvironmentType == clv1alpha2.ClassContainer && environment.Mode == clv1alpha2.ModeStandard {
-		ports = append(ports, serviceSpecTCPPort(MyDrivePortName, MyDrivePortNumber))
+	switch environment.EnvironmentType {
+	case clv1alpha2.ClassStandalone:
+
+		// Add the "MyDrive" port only if the environment is a container or standalone.
+		if environment.Mode == clv1alpha2.ModeStandard {
+			ports = append(ports, serviceSpecTCPPort(MyDrivePortName, MyDrivePortNumber))
+		}
+
+	case clv1alpha2.ClassVM, clv1alpha2.ClassContainer:
+		// Do not add the ssh port on container-based instances, since no deamon is present.
+		if environment.EnvironmentType == clv1alpha2.ClassVM {
+			ports = append(ports, serviceSpecTCPPort(SSHPortName, SSHPortNumber))
+		}
+
+		// Add the "MyDrive" port only if the environment is a container or standalone.
+		if environment.EnvironmentType == clv1alpha2.ClassContainer && environment.Mode == clv1alpha2.ModeStandard {
+			ports = append(ports, serviceSpecTCPPort(MyDrivePortName, MyDrivePortNumber))
+		}
+	default:
```

Comment:
This mixes switch/if seems overly complicated to me, and the comments do not even match anymore.

Reply:
Fixed